addons/l10n_id/models/res_bank.py

```diff
@@ -0,0 +1,67 @@
+import requests
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+QRIS_API_URL = "https://qris.online/restapi/qris/show_qris.php"
+
+class ResBank(models.Model):
+    _inherit = "res.partner.bank"
+
+    l10n_id_qris_api_key = fields.Char("QRIS API Key")
+    l10n_id_qris_mid = fields.Char("QRIS Merchant ID")
+
+    @api.model
+    def _get_available_qr_methods(self):
+        rslt = super()._get_available_qr_methods()
+        rslt.append(('id_qr', _("QRIS"), 40))
+        return rslt
+
+    def _get_error_messages_for_qr(self, qr_method, debtor_partner, currency):
+        if qr_method == 'id_qr' and self.country_code == 'ID':
+            if currency.name not in ['IDR']:
+                return _("Can't generate a QR code with a currency other than IDR.")
+            if not (self.l10n_id_qris_api_key and self.l10n_id_qris_mid):
+                return _("Please configure the QRIS API Key and Merchant ID on the bank's configuration")
+            return None
+
+        return super()._get_error_messages_for_qr(qr_method, debtor_partner, currency)
+
+    def _get_qr_vals(self, qr_method, amount, currency, debtor_partner, free_communication, structured_communication):
+        """Override
+
+        Getting content for the QR through calling QRIS API"""
+        if qr_method == "id_qr":
+            params = {
+                "do": "create-invoice",
+                "apikey": self.l10n_id_qris_api_key,
+                "mID": self.l10n_id_qris_mid,
+                "cliTrxNumber": structured_communication,
+                "cliTrxAmount": int(amount)
+            }
+            try:
+                res = requests.get(QRIS_API_URL, params=params, timeout=10)
+                res.raise_for_status()
```

Comment:
Please nooo, raising for status usually gives no useful information to the user besides a traceback and the status code. Could you make sure to try an error and see how it looks?
I would suggest checking by yourself and making a UserError with an ok-looking text instead to guide the user to the solution as much as possible

Reply:
yep I'll try some error scenarios and adjust this one